launcher/src/main/java/org/apache/spark/launcher/Main.java

```diff
@@ -101,6 +110,80 @@ public static void main(String[] argsArray) throws Exception {
     }
   }
 
+  private static void printUsage() {
```

Comment:
Yes, this one is copied from SparkSubmitArguments. SparkSubmitArguments is the scala project. I didn't find the Main.java call anything from SparkSubmitArguments in the scala file, which seems the scala version is not used. I will debug more to find out the reason.